testing/fake-editor.rs

```diff
@@ -30,8 +30,16 @@ struct Args {
 fn main() {
     let args: Args = Args::parse();
     let edit_script_path = PathBuf::from(std::env::var_os("EDIT_SCRIPT").unwrap());
-    let edit_script = String::from_utf8(std::fs::read(edit_script_path).unwrap()).unwrap();
-    for instruction in edit_script.split('\0') {
+    let edit_script = String::from_utf8(std::fs::read(edit_script_path.clone()).unwrap()).unwrap();
+
+    let mut instructions: Vec<_> = edit_script.split('\0').collect();
+    if let Some(pos) = instructions.iter().position(|&i| i == "next invocation\n") {
+        // Overwrite the edit script. The next time `fake-editor` is called, it will
+        // only see the part after the `next invocation` command.
+        std::fs::write(edit_script_path, instructions[pos + 1..].join("\0")).unwrap();
+        instructions.resize(pos, "");
```

Comment:
nit: use `instructions.truncate(pos)` instead since it seems like this would only ever truncate (never extend)

Reply:
Excellent, I missed that method.